spanner/spanner_snippets/spanner/opentelemetry/spanner_opentelemetry_usage.go

```diff
@@ -0,0 +1,182 @@
+	// Enable OpenTelemetry metrics before injecting meter provider.
+	spanner.EnableOpenTelemetryMetrics()
+
+	// Create a new tracer provider
+	tracerProvider, err := getOtlpTracerProvider(ctx, res)
+	defer tracerProvider.ForceFlush(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Register tracer provider as global
+	otel.SetTracerProvider(tracerProvider)
+
+	// Create a new meter provider
+	meterProvider := getOtlpMeterProvider(ctx, res)
+	defer meterProvider.ForceFlush(ctx)
+
+	// Inject meter provider locally via ClientConfig when creating a spanner client or set globally via setMeterProvider.
+	client, err := spanner.NewClientWithConfig(ctx, db, spanner.ClientConfig{OpenTelemetryMeterProvider: meterProvider})
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+	return nil
+}
+
+func getOtlpMeterProvider(ctx context.Context, res *resource.Resource) *sdkmetric.MeterProvider {
+	otlpExporter, err := otlpmetricgrpc.New(ctx, otlpmetricgrpc.WithInsecure())
+	if err != nil {
+		log.Fatal(err)
+	}
+	meterProvider := sdkmetric.NewMeterProvider(
+		sdkmetric.WithResource(res),
+		sdkmetric.WithReader(sdkmetric.NewPeriodicReader(otlpExporter)),
+	)
+	return meterProvider
+}
+
+func getOtlpTracerProvider(ctx context.Context, res *resource.Resource) (*sdktrace.TracerProvider, error) {
+	traceExporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithResource(res),
+		sdktrace.WithBatcher(traceExporter),
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+	)
+
+	return tracerProvider, nil
+}
+
+func newResource() (*resource.Resource, error) {
+	return resource.Merge(resource.Default(),
+		resource.NewWithAttributes(semconv.SchemaURL,
+			semconv.ServiceName("otlp-service"),
+			semconv.ServiceVersion("0.1.0"),
+		))
+}
+
+// [END spanner_opentelemetry_usage]
+
+// [START spanner_opentelemetry_gfe_metric]
+// GFE_Latency and other Spanner metrics are automatically collected
+// when OpenTelemetry metrics are enabled.
+func captureGFELatencyMetric(ctx context.Context, client spanner.Client) error {
+	stmt := spanner.Statement{SQL: `SELECT SingerId, AlbumId, AlbumTitle FROM Albums`}
+	iter := client.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	for {
+		row, err := iter.Next()
+		if err == iterator.Done {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		var singerID, albumID int64
+		var albumTitle string
+		if err := row.Columns(&singerID, &albumID, &albumTitle); err != nil {
+			return err
+		}
+	}
+}
+
+// [END spanner_opentelemetry_gfe_metric]
+
+// [START spanner_opentelemetry_capture_query_stats_metric]
+func captureQueryStatsMetric(ctx context.Context, mp metric.MeterProvider, client spanner.Client) error {
+	meter := mp.Meter(spanner.OtInstrumentationScope)
```

Comment:
Where is this OtInstrumentationScope defined ? We can hardcode the value "cloud.google.com/java" here to make it easy for users to understand.

Reply:
Yes it is a public variable that can be accessed by customers. This will prevent copy paste errors.